packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -60,7 +60,66 @@ export class WaferMap extends FoundationElement {
         values: []
     };
 
-    private readonly dataManager: DataManager | undefined;
+    private renderQueued = false;
+
+    private dataManager: DataManager | undefined;
+
+    private quadrantChanged(): void {
+        this.queueRender();
+    }
+
+    private orientationChanged(): void {
+        this.queueRender();
+    }
+
+    private maxCharactersChanged(): void {
+        this.queueRender();
+    }
+
+    private dieLabelsSuffixChanged(): void {
+        this.queueRender();
+    }
+
+    private colorsScaleModeChanged(): void {
+        this.queueRender();
+    }
+
+    private highlightedValuesChanged(): void {
+        this.queueRender();
+    }
+
+    private diesChanged(): void {
+        this.queueRender();
+    }
+
+    private colorScaleChanged(): void {
+        this.queueRender();
+    }
+
+    private queueRender(): void {
+        if (!this.renderQueued) {
+            this.renderQueued = true;
+            window.requestAnimationFrame(() => {
+                this.render();
+            });
+        }
+    }
+
+    private render(): void {
+        this.renderQueued = false;
+
+        this.dataManager = new DataManager(
+            this.dies,
+            this.quadrant,
+            { width: this.offsetWidth, height: this.offsetHeight },
```

Comment:
This will choose give size of the control when render is called but does not monitor the control for size changes.

For example the wafer-map could be used in a flex layout and should respond appropriately to css size changes.

To accomplish that we should use a ResizeObserver: https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver

In the connectedCallback it can be registered:

```ts
public override connectedCallback(): void {
  super.connectedCallback();

  this.resizeObserver = new ResizeObserver(entries => {
    const entry = entries[0];
    const {height, width} = entry.contentRect;
    this.observedWidth = width;
    this.observedHeight = height;
  });
  this.resizeObserver.observe(this);
```

In the disconnectCallback it should be unobserved.

We should add `observedWidthChanged` and `observedHeightChanged` methods to queueRender.

And then here in the render function we can instead use

```suggestion
            { width: this.observedWidth, height: this.observedHeight},
```

Reply:
Chatted offline and was explained that the existing code has built-in assumptions about rendering at specific sizes. @munteannatan will add a link to the user story for handling responding to different sizes.

Reply:
[Here](https://dev.azure.com/ni/DevCentral/_workitems/edit/2211958) we are planning to implement dynamic sizing for the wafer map component